tianshou/data/buffer.py

```diff
@@ -163,7 +163,19 @@ def update(self, buffer: 'ReplayBuffer') -> None:
             return
         i = begin = buffer._index % len(buffer)
         while True:
-            self.add(**buffer[i])
+            obs_next = None if isinstance(
```

Comment:
This is quite ugly and has several duplicates. I would rather patch `add` method if possible.